Server/mods/deathmatch/logic/packets/CEntityAddPacket.cpp

```diff
@@ -978,6 +978,42 @@ bool CEntityAddPacket::Write(NetBitStreamInterface& BitStream) const
                         BitStream.Write(currentWeaponSlot);
                     }
 
+                    // Animation
+                    if (BitStream.Can(eBitStreamVersion::AnimationsSync))
+                    {
+                        const SPlayerAnimData& animData = pPed->GetAnimationData();
+
+                        // Contains animation data?
+                        bool animRunning = !animData.blockName.empty() && !animData.animName.empty();
+
+                        // Is animation still running?
+                        float deltaTime = GetTickCount64_() - animData.startedTick;
+                        if (!animData.freezeLastFrame && animData.time > 0 && deltaTime >= animData.time)
+                        {
+                            animRunning = false;
+                            pPed->SetAnimationData({});
```

Comment:
A little bit strange to manipulate the ped's state here in the packet writing function.

Reply:
Would ``CPedSync::DoPulse`` be the appropriate place?

Reply:
I don't know enough to judge that, if you think it's a good spot then use it. I only have an issue with it when it happens in the packet writing logic.

Reply:
Done and tested